### DISCUSSION TITLE
Fixed node requirements

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#279] [General]
I am reasonably invested in this project by this point, so I'd like to ask what your requirements are for a fixed node on the network. I realize it doesn't much matter, but if you want an additional web-facing node, I'd be happy to have one.

I have two rack mounts from old projects that are under capacity. I can spare 8 gigs of RAM and at least four cores on a VM to host a node. I'd rather use the VM for cleanliness, but other server hates XCP-NG so it's bare metal. I could set up a Docker container, but I think it'd be a bad idea to run it directly in Python on the system. It's getting old, and there are a lot of stray packages.

I've got a standing domain name pointing to the servers, so it should be up for as long as I am.  Uptime's not perfect, we experience power failures a few times a year, but otherwise it's 24/7. Current uptime 155 days.

Do you think it's worth setting up another reference node, or do you think the current examples are sufficient?

**Comment by ghost** [comment]
Reticulum runs today on anything with Python support, including single-board computers. Your VM setup would be overkill to dedicate as simply a Reticulum node, as the network I/O bottleneck would never demand such power or space for managing its connections. You could serve as a highly available LXMF router capable of storing large caches of messages for users, though, but I have no experience with that protocol yet to know.

Reducing hardware requirements to establish resilient connectivity showcases how you might think differently about contributing to the project; there exists a testnet you could connect to over the Internet (which is perhaps how you'd offer the aforementioned hypothetical LXMF service), but instead of contributing resources to a high-level global effort, what can your VM offer to connections immediately outside its interfaces (ok, a few hops considering vNICs are involved)? To defy TCP/IP is an enormous endeavor, and nature teaches that scalability is achieved not by building big but growing from small.

In the past month getting to know this community, I've learned of one group aiming to build a network across my state, and I've been inspired to reach out to a local city wireless project in the hope of introducing more advanced meshing tech like Reticulum, at least to their attention. The project is still nascent, and I think the biggest contributions would be in understanding its technology and sharing it with others around you. You then get the benefits of mesh networking even in typically offline scenarios, plus the assurance that it's actually secure and private without selling your soul to a big tech company.

**Reply by faragher** [reply]
I've discussed local networking with another user a few hours from here. Right now the infrastructure isn't close enough to make a serious effort, and my current RNode experimentation precludes me from linking my radios to the test net unless people really love hundreds of random announces from addresses that last for one or two transmissions. At this point I think the experimentation and documentation is the more important effort, so that's how I'm dedicating my radios unless there's actual local demand,

But I believe I was unclear. I'm not looking for minimum requirements for personal use, I'm asking for a dedicated and well-known propagation node, similar to the Dublin node. If having another TCP/IP facing node isn't helpful, that's a different discussion.

**Comment by markqvist** [comment]
I would LOVE to see some more permanent nodes on the public internet for people to connect to. The less infrastructure I have to run, the more I can focus on just the actual development and research, which is where my time is best spend.

The requirements for running one are really low. If you can spare a server like what you mention, even just a VM with part of that, it'd easily be able to handle *thousands* of connections and route traffic between a lot of different reticules.

So yes, if you're up for it, that would be very much appreciated, and I'd love to add it to the list of public connection points. I can also point a DNS record to it under the ``connect.reticulum.network`` name space.

**Reply by faragher** [reply]
Done. I can shoot you the IP. but it's not fixed so much as "sticky." If we need to replace the router again, it'll change. It's your choice if you want to add it and maybe need to fiddle with it every few years. (We've had to change it around 6 times in the last ten years, but I have faith in this router)

Currently listening on betweentheborders.com:4242 - set it up as a service, node is enabled, announce and sync at six hours, maximum messages 8 (the default). Should I change those? I think they're client options.  I upped the message store to 12 gigs. Anything else that I should look at or double check?

Propagation node is 2570d535b9d6ef58a4c3781ddb60ea92 if you want to make sure it's actually being seen on the nest net. 

Be advised that when it's in a stable state, it will be connected to a base station RNode, although I doubt it'll be much more than me using it. Bunch'a luddites in this town, I tell ya. :D

**Reply by faragher** [reply]
Oh, and it's got uplinks to both Dublin and Frankfurt in the config.

**Reply by markqvist** [reply]
That is awesome! Thank yoxu so much for setting this up! I can resolve paths to `2570d535b9d6ef58a4c3781ddb60ea92` from here, and I can see that my own propagation nodes have already peered with it :)

Instead of having to maintain extra domain names and update the IPs, I think I'll just add "community nodes" section to the relevant testnet sections of the docs instead, and point it directly to your domain, if that is all good with you.

**Reply by faragher** [reply]
Sounds perfect. Glad I can help.

**Comment by markqvist** [comment]
And also, as @jonarmani says, running a LXMF propagation node on the system, with just a few gigabytes of storage set aside for it, would be really nice too :)
